keylime/ima/ima.py

```diff
@@ -219,6 +219,7 @@ def _validate_ima_buf(
             failure = _validate_ima_ng(exclude_regex, allowlist, digest, path, hash_types="keyrings")
             if not failure:
                 # Add the key only now that it's validated (no failure)
+                assert ima_keyrings is not None
```

Comment:
Under normal conditions `ima_keyrings` is never None when an IMA validator is called, but for now I would rather check it explicitly instead of an assertion.

We definitely need to refactor the agent state at some point into a single object.

Reply:
https://github.com/keylime/keylime/pull/1139/commits/e6b2344b0d1ab9b4c0d6ccfbb8399e2fe5dccf5c